hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -487,20 +486,23 @@ observeCommitTx networkId initials tx = do
     , (commitIn, toUTxOContext commitOut, dat)
     )
  where
-  findInitialTxIn =
-    case filter (`elem` initials) (txIns' tx) of
-      [input] -> Just input
-      _ -> Nothing
+  initialScript = fromPlutusScript Initial.validatorScript
 
-  decodeInitialRedeemer =
-    findRedeemerSpending tx >=> \case
-      Initial.Abort ->
+  findCommittedTxIn =
+    case redeemersSpending of
+      [fromScriptData -> Just Initial.Abort] ->
         Nothing
-      Initial.Commit{committedRef} ->
+      [fromScriptData -> Just Initial.Commit{committedRef}] ->
         Just (fromPlutusTxOutRef <$> committedRef)
+      _ ->
+        Nothing
 
-  commitAddress = mkScriptAddress @PlutusScriptV1 networkId commitScript
+  redeemersSpending = foldRedeemers tx [] $ \tag _ dat _ rs ->
+    case tag of
+      Spend -> dat : rs
+      _ -> rs
```

Comment:
IMO the `findRedeemerSpending` was a bit clearer before

Reply:
Question: This code seems more complicated than it was before without adding much value. Am I missing something?